### PR TITLE
Cart page

### PR DESCRIPTION
**Pull request by abhishektanwar**
Files to check:

- src/screens/Cart/index.jsx
- src/screens/Cart/cart.css
- src/components/TicketContainer/index.jsx
- src/components/ProductCard/index.jsx (line 16 - 98, horizontal variant)

### CODE REVIEW

File: src/screens/Cart/cart.css [hunk 1]
```diff
@@ -0,0 +1,55 @@
+.total-ticket-container{
+	background-color: var(--bg-color-primary);
+	height: fit-content;
+	width: 45rem;
+	min-width:40rem;
+	position: sticky;
+	top: 10.4rem;
+	color: var(--black-color);
```
Comment (zoyatanweer) [inline src/screens/Cart/cart.css]: you can write background color and color together, this way the user shouldn't have to go through all the lines in between.

File: src/screens/Cart/cart.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	border-bottom: 2px solid var(--border-color);
+}
+
+.margin-top-bottom-10{
+	margin: 10px 0;
+}
+
+/* .total-ticket-container .card-text-container{
+	width: 100%;
+} */
```
Comment (zoyatanweer) [inline src/screens/Cart/cart.css]: remove commented code if not required

File: src/components/ProductCard/product-card.css [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  flex-direction: column;
+}
+
+.image-container{
+  height: 30rem;
+}
+
+.vertical-card{
+  max-width: 280px;
+  min-width: 280px;
```
Comment (zoyatanweer) [inline src/components/ProductCard/product-card.css]: these two can be in rem

File: src/components/ProductCard/product-card.css [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.favorite-btn{
+  position: absolute;
+  top: 15px;
+  right: 10px;
+  cursor: pointer;
+  /* background: chartreuse; */
```
Comment (zoyatanweer) [inline src/components/ProductCard/product-card.css]: remove commented code if not required.

File: src/components/ProductCard/product-card.css [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+.primary-card-badge{
+  color: rgba(255, 255, 255, 0.952);
+}
+
+.quantity-selector > select{
+  border: none;
+  background-color: transparent;
+  margin-right: 3rem;
+  cursor: pointer;
+}
```
Comment (zoyatanweer) [inline src/components/ProductCard/product-card.css:52]: I like the way how you used the 'direct child selector' here. Nice.

File: src/components/ProductCard/product-card.css [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+.cart-action-btn-container{
+  display: flex;
+  align-items: center;
+}
+
+.cart-action-btn-container .secondary-button{
+  color:var(--primary-color)
```
Comment (zoyatanweer) [inline src/components/ProductCard/product-card.css]: semicolon not needed here?

File: src/components/ProductCard/product-card.css [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.vertical-card .card-text-container{
+  width: unset;
+  padding: 0.6rem 1.6rem;
+}
+
+.horizontal-card .image-container{
+  margin: auto 0.6rem;
```
Comment (zoyatanweer) [inline src/components/ProductCard/product-card.css]: isn't it small value? it can be in px here I think.

**Review by zoyatanweer — APPROVED** [review]
code is clean and easily understandable. Good to merge with suggested changes. Great work Abhishek.

**Review by sudipkundu999 — CHANGES_REQUESTED** [review]
The UI looks good and so is the code.

The naming of files is inconsistent.
In the ProductListing component, you've named the file as `ProductListing.jsx` while in other places you've named it as `index.jsx`. It would become difficult to work with multiple files with the same names doing different things.

As a rule of thumb:
When you're making a component eg. TicketContainer name the file as `TicketContainer.jsx` and not `index.jsx`

Also, use the `index.js` file for import-export, this will avoid multiple import lines
eg. in App.js (line 5,7,8) for 3 screens there are 3 import lines.
As the number of screens increases import lines will also increase.

If you keep an index.js in screens folder then this will reduce to 
`import {ProductListing, Wishlist, Cart} from "./screens"`

**Comment by abhishektanwar** [comment]
> As a rule of thumb:
When you're making a component eg. TicketContainer name the file as TicketContainer.jsx and not index.jsx

Yes I understood what you are trying to convey here but to me, it makes more sense to have `index.js` file as it helps me in understanding what is the starter file and where to begin from in a folder. 

>Also, use the index.js file for import-export, this will avoid multiple import lines
eg. in App.js (line 5,7,8) for 3 screens there are 3 import lines.
As the number of screens increases import lines will also increase. 

Coming to keeping an `index.js` file in 'screens` directory, that can still be done. I'll make this older structure change once all the current branches are merged

File: src/screens/Cart/cart.css (hunk 2 above)
Comment (abhishektanwar) [reply src/screens/Cart/cart.css]: removed

File: src/components/ProductCard/product-card.css (hunk 3 above)
Comment (abhishektanwar) [reply src/components/ProductCard/product-card.css]: fixed.

File: src/components/ProductCard/product-card.css (hunk 4 above)
Comment (abhishektanwar) [reply src/components/ProductCard/product-card.css]: removed

**Review by sudipkundu999 — APPROVED** [review]
The reasoning for keeping `index.js` as file name makes sense now.
Good work. Good to merge.

**Pull request merged**